dirs/dirs.go

```diff
@@ -521,6 +608,9 @@ func SetRootDir(rootdir string) {
 	for _, c := range callbacks {
 		c(rootdir)
 	}
+
+	// If the root directory changes we also need to reset snapHomeDirs.
```

Comment:
I would move this above the callbacks so if they actually inspect the computed data, they would see the correct values.